pkg/aws/errors/utils.go

```diff
@@ -0,0 +1,32 @@
+package errors
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/codes"
+)
+
+// GetMCMErrorCodeForCreateMachine takes the error returned from the EC2API during the CreateMachine call and returns the corresponding MCM error code.
+func GetMCMErrorCodeForCreateMachine(err error) codes.Code {
+	awsErr := err.(awserr.Error)
+	return mapErrorCodeForCreateMachine(awsErr.Code())
+}
+
+func mapErrorCodeForCreateMachine(errCode string) codes.Code {
```

Comment:
we can move this code in `GetMCMErrorCodeForCreateMachine()` as its not much code overall